common/namespace/registry.go

```diff
@@ -135,6 +135,7 @@ type (
 		GetNamespace(name Name) (*Namespace, error)
 		GetNamespaceWithOptions(name Name, opts GetNamespaceOptions) (*Namespace, error)
 		GetNamespaceByID(id ID) (*Namespace, error)
+		RefreshSingleCacheKey(namespaceName Name, namespaceId ID) error
```

Comment:
I'd also return the refreshed namespace entry as the response. I guess in most cases, that's what the caller want and they don't have to make a follow up GetNamespaceByID/Name() call.